src/main/java/com/google/sps/TextDetectionAPIImpl.java

```diff
@@ -82,12 +82,15 @@ private BatchAnnotateImagesResponse detectTextFromImage(AnnotateImageRequest req
             "An error occurred while identifying the text from the image\n"
                 + identifiedText.getError().getMessage());
       }
-      // Future Scope: Determine sentence's height by subtracting lower boundary (lower y-axis position) and 
-      // upper boundary (upper y-axis position) to help in handwritten recognition when sentences height ration will vary
+      // Future Scope: Determine sentence's height by subtracting lower boundary (lower y-axis
+      // position) and upper boundary (upper y-axis position) to help in handwritten recognition
+      // when sentences height ration will vary
       for (EntityAnnotation annotation : identifiedText.getTextAnnotationsList()) {
```

Comment:
Yes, I did check this out earlier. The reason for not using [FullTextAnnotation](https://cloud.google.com/vision/docs/reference/rest/v1p4beta1/AnnotateImageResponse#textannotation) is because it has higher time complexity (with 3 loops, if we consider taking paragraphs) whereas the code I wrote only has one loop to get the detected text (word) O(n). It is also recommended in CloudVisionAPI documentation that FullTextAnnotation should be used for scanning through documents with dense text [Reference :]( https://cloud.google.com/vision/docs/ocr?apix_params=%7B%22resource%22%3A%7B%7D%7D#optical_character_recognition_ocr).
 
In case of shopping list, I declared a feature called "TEXT DETECTION" to make it more time efficient O(n). In the future, my design could be easily changed to add another feature switch for "DOCUMENT TEXT DETECTION" if we need to cover densely populated documents or handwritten images.

Reply:
Yes, I understand, thank you for the explanation.